### PR TITLE
Pundit index

### PR DESCRIPTION
**Pull request by TofuSupreme**
Please note:
For some strange reason I was unable push this from my terminal to git hub, so I've implemented the files via GH. It may or may not. (But it works on my local computer)
I've made changes so that any user (for the time being) is able to access the index page. 
Please check BEFORE pushing to master! I'll try to talk to a TA if this doesn't work.

### CODE REVIEW

**Review by spioupiou — COMMENTED** [review]
I am entirely sure but I think index is a special case where we need policy_scope. I'll have a fresh look tomorrow morning :) 

Screen Shot 2022-01-25 at 22 51 37

**Comment by carlnoval** [comment]
Yeah pundit is very particular on index action 😅 thanks for linking that reference @spioupiou.

I'll also be pasting more resources here so its easier for us to implement/review Pundit.
- [kitt lecture Source docs](https://kitt.lewagon.com/camps/718/lectures/05-Rails%2F07-Airbnb-Facebook-connect)
- [kitt video, index starts at 56:34](https://kitt.lewagon.com/camps/718/lectures/05-Rails%2F07-Airbnb-Facebook-connect)

**Comment by TofuSupreme** [comment]
Yes, you're absolutely right @spioupiou. Instead of your approach, I've changed the exception to the verify_scope_policy in the application_controller as opposed to the Kondos Controller. 

```
class ApplicationController < ActionController::Base
  before_action :authenticate_user!
  before_action :configure_permitted_parameters, if: :devise_controller?

  include Pundit

  # Pundit: white-list approach.
  after_action :verify_authorized, except: :index, unless: :skip_pundit?
  after_action :verify_policy_scoped, except: :index, unless: :skip_pundit?
```

But I'll try @spioupiou's method on my local terminal and see if that works too. :) 

@carlnoval Thanks for the resources!

File: app/controllers/kondos_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class KondosController < ApplicationController
   def index
-    @kondos = Kondo.order(created_at: :desc)
+    authorize @kondos = Kondo.order(created_at: :desc)
```
Comment (spioupiou) [inline app/controllers/kondos_controller.rb:3]: Maybe it works this way but might be safer to follow the class example and use @kondos = policy_scope(Kondo).order(created_at: :desc) :)

Comment (spioupiou) [reply app/controllers/kondos_controller.rb:3]: but yeah if you skip the pundit for index we probably don't need to use policy_scope at all :) In that case I would maybe not use authorize either actually... my knowledge of pundit is way too limited haha

**Pull request closed without merging**